rust/nasl-interpreter/src/built_in_functions/ssh.rs

```diff
@@ -0,0 +1,2107 @@
+            };
+            if verbose {
+                ctx.logger()
+                    .info(&format!("Available methods:\n{:?}", methods));
+            }
+
+            if methods == AuthMethods::NONE {
+                return Ok(NaslValue::Number(0));
+            }
+
+            /* Check whether a password has been given.  If so, try to
+            authenticate using that password.  Note that the OpenSSH client
+            uses a different order it first tries the public key and then the
+            password.  However, the old NASL SSH protocol implementation tries
+            the password before the public key authentication.  Because we
+            want to be compatible, we do it in that order. */
+            if password.is_some() && methods.contains(AuthMethods::PASSWORD) {
+                match session.session.userauth_password(None, password) {
+                    Ok(AuthStatus::Success) => {
+                        return Ok(NaslValue::Number(0));
+                    }
+                    Ok(_) => {
+                        if verbose {
+                            ctx.logger().info(&format!(
+                                "SSH password authentication failed for session {}",
+                                session_id
+                            ));
+                        }
+                    }
+                    Err(_) => {
+                        return Err(FunctionErrorKind::Diagnostic(
+                            format!(
+                                "Failed setting user authentication for SessionID {}",
+                                session_id
+                            ),
+                            Some(NaslValue::Null),
+                        ));
+                    }
+                };
+            }
+
+            /* Our strategy for kbint is to send the password to the first
+            prompt marked as non-echo.  */
+            if password.is_some() && methods.contains(AuthMethods::INTERACTIVE) {
+                loop {
+                    match session.session.userauth_keyboard_interactive(None, None) {
+                        Ok(AuthStatus::Info) => {
+                            let info = match session.session.userauth_keyboard_interactive_info() {
+                                Ok(i) => i,
+                                Err(_) => {
+                                    return Err(FunctionErrorKind::Diagnostic(
+                                        format!(
+                                            "Failed setting user authentication for SessionID {}",
+                                            session_id
+                                        ),
+                                        Some(NaslValue::Null),
+                                    ));
+                                }
+                            };
+                            if verbose {
+                                ctx.logger().info(&format!("SSH kbdint name={}", info.name));
```

Comment:
because this is use an ENV variable to switch it on, different from the general debug level. It will only be available if it is set via, which also turns the libssh debug log on     
```
let verbose = env::var("OPENVAS_LIBSSH_DEBUG")